SimCalorimetry/HGCalSimProducers/src/HGCHEbackDigitizer.cc

```diff
@@ -11,27 +11,312 @@
+}
+
+double HGCHEbackSignalScaler::getFluenceValue(const int layer, const std::array<double, 8>& radius)
+{
+  double cellFluence = std::pow(10, doseMap_[layer].f_ + doseMap_[layer].g_*radius[0] + doseMap_[layer].h_*radius[1] + doseMap_[layer].i_*radius[2] + doseMap_[layer].j_*radius[3]); //dose in grey
+  return cellFluence;
+}
+
+std::pair<float, float> HGCHEbackSignalScaler::scaleByDose(const HGCScintillatorDetId& cellId,  const std::array<double, 8>& radius)
+{
+  if(doseMap_.empty())
+    return std::make_pair(1., 0.);
+
+  int layer = cellId.layer();
+  double cellDose = getDoseValue(layer, radius); //in kRad
+  constexpr double expofactor = 1./199.6;
+  double scaleFactor = std::exp( -std::pow(cellDose, 0.65) * expofactor );
+
+  double cellFluence = getFluenceValue(layer, radius); //in 1-Mev-equivalent neutrons per cm2
+
+  constexpr double factor = 2. / (2*1e13); //SiPM area = 2mm^2
+  double noise = 2.18 * sqrt(cellFluence * factor);
+
+  if(verbose_)
+  {
+    LogDebug("HGCHEbackSignalScaler") << "HGCHEbackSignalScaler::scaleByDose - Dose, scaleFactor, fluence, noise: "
+                                      << cellDose << " " << scaleFactor << " "
+                                      << cellFluence << " " << noise;
+
+    LogDebug("HGCHEbackSignalScaler") << "HGCHEbackSignalScaler::setDoseMap - layer, a, b, c, d, e, f: "
+                                      << layer << " "
+                                      << doseMap_[layer].a_ << " "
+                                      << doseMap_[layer].b_ << " "
+                                      << doseMap_[layer].c_ << " "
+                                      << doseMap_[layer].d_ << " "
+                                      << doseMap_[layer].e_ << " "
+                                      << doseMap_[layer].f_;
+  }
+
+  return std::make_pair(scaleFactor, noise);
+}
+
+float HGCHEbackSignalScaler::scaleByArea(const HGCScintillatorDetId& cellId, const std::array<double, 8>& radius)
+{
+  float edge;
+  if(cellId.type() == 0)
+  {
+    constexpr double factor = 2 * M_PI * 1./360.;
+    edge = radius[0] * factor; //1 degree
+  }
+  else
+  {
+    constexpr double factor = 2 * M_PI * 1./288.;
+    edge = radius[0] * factor; //1.25 degrees
+  }
+
+  float scaleFactor = refEdge_ / edge;  //assume reference 3cm of edge
+
+  if(verbose_)
+  {
+    LogDebug("HGCHEbackSignalScaler") << "HGCHEbackSignalScaler::scaleByArea - Type, layer, edge, radius, SF: "
+                                      << cellId.type() << " "
+                                      << cellId.layer() << " "
+                                      << edge << " "
+                                      << radius[0] << " "
+                                      << scaleFactor << std::endl;
+  }
+
+  return scaleFactor;
+}
+
+std::array<double, 8> HGCHEbackSignalScaler::computeRadius(const HGCScintillatorDetId& cellId)
+{
+  GlobalPoint global = hgcalGeom_->getPosition(cellId);
+
```

Comment:
@deguio , sorry for the late comment but it seems std::pow is not needed in this method, it is only overhead.

Reply:
hi @civanch 
are you suggesting to use plain multiplications instead of `std::pow`?

Reply:
@deguio , yes, exactly

Reply:
Modern compilers optimize `pow(x,2)` to be the same as `x*x`, but not sure about other exponents

Reply:
Quick tests with godbolt indicate that GCC 7/8/9 and clang 7/8 compile `pow(x, 3)` to an actuall call to `pow()`.

Reply:
ok. so what's the suggestion? change to plain multiplication for the third degree exponents?

Reply:
Best would be e.g. `double radius_m100_3 = radius_m100_2*radius_m100;`